[BUG] Workflow execution fails on Cloudflare workers

### Describe the Bug

Workflow execution fails on Cloudflare workers with the error (from the cf dashboard):

<img width="1516" height="337" alt="Image" src="https://github.com/user-attachments/assets/2a4c635d-25e9-4d7a-8bdb-e972802f0239" />

<img width="1516" height="597" alt="Image" src="https://github.com/user-attachments/assets/438e7e0c-a7c1-40bc-a7da-ea05e22d979a" />

Local run with `wrangler dev` inside `.mastra/output`
```bash
[wrangler:info] POST /api/workflows/weatherWorkflow/create-run 200 OK (9ms)
✘ [ERROR] Error executing step: Error: Cannot perform I/O on behalf of a different request. I/O objects (such as streams, request/response bodies, and others) created in the context of one request handler cannot be accessed from a different request's handler. This is a limitation of Cloudflare Workers which allows us to improve overall performance. (I/O type: RefcountedCanceler)

      at DefaultExecutionEngine.execute
  (file:///home/kantha/mastra-cf-d1-test/.mastra/output/mastra.mjs:23163:54)
      at async Run.start
  (file:///home/kantha/mastra-cf-d1-test/.mastra/output/mastra.mjs:25352:20)
      at async startAsyncWorkflowHandler$1
  (file:///home/kantha/mastra-cf-d1-test/.mastra/output/index.mjs:6877:20)
      at async startAsyncWorkflowHandler
  (file:///home/kantha/mastra-cf-d1-test/.mastra/output/index.mjs:14718:20)
      at async dispatch
  (file:///home/kantha/mastra-cf-d1-test/.mastra/output/index.mjs:359:17)
      at null.<anonymous> (async
  file:///home/kantha/mastra-cf-d1-test/.mastra/output/.wrangler/tmp/dev-P1R7yY/index.js:43738:9)
      at async dispatch
  (file:///home/kantha/mastra-cf-d1-test/.mastra/output/index.mjs:359:17)
      at async dispatch
  (file:///home/kantha/mastra-cf-d1-test/.mastra/output/index.mjs:359:17)
      at async dispatch
  (file:///home/kantha/mastra-cf-d1-test/.mastra/output/index.mjs:359:17)
      at async dispatch
  (file:///home/kantha/mastra-cf-d1-test/.mastra/output/index.mjs:359:17)

```


### Steps To Reproduce

1. Create a new project with `npx create-mastra@latest`
2. Add `CloudflareDeployer` to the mastra instance
   ```typescript
   export const mastra = new Mastra({
     deployer: new CloudflareDeployer({
       projectName: "mastra-cf-d1-test",
     }),
     workflows: { weatherWorkflow },
     agents: { weatherAgent },
     logger: new PinoLogger({
       name: 'Mastra',
       level: 'info',
     }),
     telemetry: {
       enabled: false,
     }
   });
   ```
3. Run `mastra build`
4. Run `wrangler dev` from `.mastra/output`
5. Run,
    ```bash
    curl -X 'POST'  'http://localhost:8787/api/workflows/weatherWorkflow/create-run' \
      -H 'accept: */*'
   ```
   works,
   ```bash
   {"runId":"e9f2f05e-09c5-4dd2-8242-cdd1ce7e9cf9"}
   ```
6. Run,
    ```bash
    curl -X 'POST'  'http://localhost:8787/api/workflows/weatherWorkflow/start-async?runId=e9f2f05e-09c5-4dd2-8242-cdd1ce7e9cf9' \
       -H 'accept: */*' \
       -d '{ "inputData": {"city": "San Francisco" }'
    ```
    errors,
    ```bash
    {"error":"Cannot read properties of undefined (reading 'result')"}
    ```

### Expected Behavior

The workflow should execute successfully.

### Environment Information

- dependencies:
  ```json
  "@ai-sdk/openai": "^1.3.23",
  "@mastra/cloudflare-d1": "^0.12.3",
  "@mastra/core": "^0.11.1",
  "@mastra/deployer-cloudflare": "^0.11.1",
  "@mastra/libsql": "^0.11.2",
  "@mastra/loggers": "^0.10.4",
  "@mastra/memory": "^0.11.5",
  "dotenv": "^17.2.1",
  ```
- `OPENAI_API_KEY` set in `.env`
- OS: PopOS 22.04


### Verification

- [x] I have searched the existing issues to make sure this is not a duplicate
- [x] I have included sufficient information for the team to reproduce and understand the issue

--- a/.changeset/many-parts-design.md
+++ b/.changeset/many-parts-design.md
@@ -1,0 +1,5 @@
+---
+'@mastra/core': patch
+---
+
+lazy initialize Run's `AbortController`

--- a/packages/core/src/workflows/workflow.ts
+++ b/packages/core/src/workflows/workflow.ts
@@ -1210,7 +1210,7 @@ export class Workflow<
     const nestedAbortCb = () => {
       abort();
     };
-    run.abortController?.signal.addEventListener('abort', nestedAbortCb);
+    run.abortController.signal.addEventListener('abort', nestedAbortCb);
     abortSignal.addEventListener('abort', async () => {
       run.abortController.signal.removeEventListener('abort', nestedAbortCb);
       await run.cancel();
@@ -1337,7 +1337,7 @@ export class Run<
   TInput extends z.ZodType<any> = z.ZodType<any>,
   TOutput extends z.ZodType<any> = z.ZodType<any>,
 > {
-  public abortController: AbortController;
+  #abortController?: AbortController;
   protected emitter: EventEmitter;
   /**
    * Unique identifier for this workflow
@@ -1406,7 +1406,14 @@ export class Run<
     this.emitter = new EventEmitter();
     this.retryConfig = params.retryConfig;
     this.cleanup = params.cleanup;
-    this.abortController = new AbortController();
+  }
+
+  public get abortController(): AbortController {
+    if (!this.#abortController) {
+      this.#abortController = new AbortController();
+    }
+
+    return this.#abortController;
   }
 
   /**

